CONCEPTS.md

```diff
@@ -0,0 +1,62 @@
+# Concepts
+
+## Table of Contents
+- [Metric](#metric)
+- [Tag Key and Tag Value](#tag-key-and-tag-value)
+- [Timestamp](#timestamp)
+- [Timeseries](#timeseries)
+- [Gauge](#gauge)
+- [Counter](#counter)
+- [Timer](#timer)
+- [Metric Registry](#metric-registry)
+- [Reporter](#reporter)
+- [Time Window](#time-window)
+- [Monoid](#monoid)
+
+### Metric
+A quantifiable characteristic of an application that is being measured. Metrics have numeric values that are either 64-bit signed integers or 64-bit double-precision floating-point values. Metrics have names that are strings and should meaningfully describe the quantity being measured. Some examples -
+* `webserver.rps` the number of requests per second being processed by a web server
+* `webserver.request.latency` time taken (latency, cumulative or percentiles) to process requests by a web server
+
+### Tag Key and Tag Value
+A tag (aka dimension) is a pair that consists of a key and a value. Tags describe the context for which a metric is being recorded. For example, `host=webserver1.domain.com`. Here, `host` is a tag key and `webserver1.domain.com` is the tag value. Other examples could include `url`, `resource-path`, `response-code`, `application-id`, `region`, or `availability-zone`.
+
+In addition to being more descriptive, tags allow identification of metrics at a fine-grained level. These could be used for flexible analysis via monitoring and reporting systems whereby one could get data summaries along any combination of tags along which the metrics were recorded. For example, total number of requests handled by all web servers in a region.
+
+### Timestamp
+For the purposes of this library, a timestamp represents the end of a time interval for which a metric has been recorded and reported. If a metric is reported every 10 seconds, the timestamp should represent the end of a given 10-second interval, e.g., 10, 20, or 30 seconds past the minute. If metrics are reported immediately upon receipt by the library, the timestamp should represent the moment when the update was recorded.
+
+### Timeseries
+A time series is a sequence of timestamp and numeric-value pairs where consecutive timestamps are monotonically increasing. For our purposes, the identity of a time series is the unique combination of metric name and associated tags, and the series itself consists of the timestamps for which values were reported and values of the metric at those timestamps.
```

Comment:
Should we mention timeseries can be scarce, i.e. no values on a set of intervals but can have gaps?